### DISCUSSION TITLE
Marker Import from Vimeo Imports Unwanted Markers

### DISCUSSION THREAD

**Discussion by moosefuel** [latenitefilms/MarkerToolbox#2] [Q&A]
Hi Chris & Co.,

When I import a .csv file from Vimeo into my timeline, I get some pretty unusual behaviour. I'm not sure if it is importing markers from old versions, or just importing them out of order. I hope this is the place to post this. 

If you look at [the screen recording here](https://vimeo.com/moosefuel/review/820631582/29d529a3e6), you can see that...

1. I download the .csv file from the Vimeo review page,
2. open Marker Toolbox and select the downloaded .csv file,
3. drag the green xml icon onto my timeline
4. peruse the markers, some of which don't seem to match the actual comments on the review page,
5. open up the .csv file in Apple Numbers to find out where the extra markers are from

I don't understand very well the syntax of these .csv files, maybe this is expected behaviour and I just don't understand? 

I am attaching the .csv file also.
[review_notes_808783529.csv](https://github.com/latenitefilms/MarkerToolbox/files/11313615/review_notes_808783529.csv)

Thanks for the help.

**Comment by latenitefilms** [comment]
Thanks for the `.CSV` and screen recording!

Yes, currently Marker Toolbox will import all the markers in the CSV.

There's no information about the length of the video in the CSV, so the Compound Clip is only as long as the last marker.

The reason you have some green markers and some red markers is that the green ones have been "resolved" in Vimeo.

Maybe we need to add a setting to not import resolved markers? I've added to issue #3.

I'll have to do some more testing, but I was under the impression that when you save a CSV from Vimeo it only exports the current version - but apparently not! It looks like the CSV contains ALL versions.

The quick fix/workaround would be to open the CSV file in a Text Editor (such as [BBEdit](https://www.barebones.com/products/bbedit/)) and remove the lines which contain the previous versions.

As a proper fix, we could add some kind of user interface for selecting the version you want to import.

The tricky part with that is that based on your screen recording:

<img width="289" alt="image" src="https://user-images.githubusercontent.com/22286696/234228203-9ff26ec9-66dd-41dc-885e-38816d7c8f2b.png">

...a lot of your versions have the same name (i.e. "2023 Revolution 4.0 Movie"). However, it does look like the CSV has a blank line between versions, which we can use - it would just list "2023 Revolution 4.0 Movie" 4 times (and "Unknown file name").

I've added to issue #4.

In the meantime, here's a modified CSV file with only the latest version:

[review_notes_808783529_latest_version.csv](https://github.com/latenitefilms/MarkerToolbox/files/11320790/review_notes_808783529_latest_version.csv)

**Comment by moosefuel** [comment]
> I was under the impression that when you save a CSV from Vimeo it only exports the current version - but apparently not!

Me too, my friend. I think this is new behaviour from Vimeo. Thanks for the modified CSV, I was also able to create my own by ordering it by date and deleting old comments. A drop-down menu or something to select just the most recent version would be ideal.

The reason my files all have the same name is simply because I use the Vimeo Uploader, and like a blind monkey I keep just pressing the buttons to update the existing video with the current version of the timeline, without any system of versioning. I could develop some new habits if that would help.

Thanks.
